### ISSUE TITLE
No auto completion with new version of Jupyter

### ISSUE THREAD

**Issue by 0liu** [nnicandro/emacs-jupyter#309]
Thanks a lot for making this great package! It really saves me from coding in a web browser.
I have two conda envrionments. The older one is installed based on Anaconda 2020.11 and the newer one is from conda-forge. completion-at-point works well in the older conda env but always throws "No matches" in the new conda env. I guess the package has not been tested on the new version of Jupyter? If that's the case, can you please make it work for the new version? Here are the versions of the two envs:

**Old env**

```
Name                    Version                   Build  Channel
jupyter                   1.0.0                    py38_7  
jupyter-contrib-core      0.3.3                    pypi_0    pypi
jupyter-contrib-nbextensions 0.5.1                    pypi_0    pypi
jupyter-highlight-selected-word 0.2.0                    pypi_0    pypi
jupyter-latex-envs        1.4.6                    pypi_0    pypi
jupyter-nbextensions-configurator 0.4.1                    pypi_0    pypi
jupyter_client            6.1.7                      py_0  
jupyter_console           6.2.0                      py_0  
jupyter_core              4.6.3                    py38_0  
jupyterlab                2.2.6                      py_0  
jupyterlab_pygments       0.1.2                      py_0  
jupyterlab_server         1.2.0                      py_0  
ipykernel                 5.3.4            py38h5ca1d4c_0  
ipython                   7.19.0           py38hb070fc8_0  
ipython_genutils          0.2.0                    py38_0  
ipywidgets                7.5.1                      py_1  
```

**New env**
```
Name                    Version                   Build  Channel
jupyter                   1.0.0            py38h578d9bd_4    conda-forge
jupyter_client            6.1.11             pyhd8ed1ab_1    conda-forge
jupyter_console           6.2.0                      py_0    conda-forge
jupyter_core              4.7.0            py38h578d9bd_0    conda-forge
jupyter_server            1.2.2            py38h578d9bd_0    conda-forge
jupyterlab                3.0.5              pyhd8ed1ab_0    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_server         2.1.2              pyhd8ed1ab_0    conda-forge
jupyterlab_widgets        1.0.0              pyhd8ed1ab_1    conda-forge
ipykernel                 5.4.2            py38h81c977d_0    conda-forge
ipython                   7.12.0           py38h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.3              pyhd3deb0d_0    conda-forge
```

**Comment by timlod** [comment]
I believe this issue is caused not by `jupyter` or `emacs-jupyter`, but `jedi`. Did you use `jedi==0.18` in your new environment by any chance? That one isn't compatible with IPython, and hence autocomplete is unavailable. A fix is pinning jedi to `jedi==0.17.2`.

**Comment by 0liu** [comment]
You are absolutely correct. According to [ipython #12740](https://github.com/ipython/ipython/issues/12740), they are incompatible until ipython 7.19.
